feat(curriculum): add musical instrument workshop (#61750)
freeCodeCamp/freeCodeCamp: curriculum/challenges/english/25-front-end-development/workshop-musical-instrument-inventory/6895e356fd877efd6e3657ee.md

---
id: 6895e356fd877efd6e3657ee
title: Step 8
challengeType: 20
dashedName: step-8
---

# --description--

Methods are functions defined inside a class that can perform actions using the class's data. 

Here is how you create methods:

```python
def method_name(parameter1, parameter2,...):
    # Method contents will go here

> ```

Reviewer: this seems a bit late, you have already asked to create an `__init__` method and there was no example at the time

> Methods are functions defined inside a class that can perform actions using the class's data. 
> 
> Here is how you create methods:
> 
> ```python
> def method_name(parameter1, parameter2,...):
>     # Method contents will go here
> ```

Reviewer 2: This is exactly what has been done in step 2. I don't think we need to repeat the same instructions.
```suggestion
```

Author: I forgot to remove these.

> Inside the `MusicalInstrument` class, define a method named `play` that takes only the `self` parameter. Add `pass` in the body. The `self` parameter allows methods to access the specific instance's attributes.

Reviewer 2: ```suggestion
Inside the `MusicalInstrument` class, define another method named `play` with only the `self` parameter. Add `pass` in the body. The `self` parameter allows methods to access the specific instance's attributes.
```

# --hints--

> You should define a method named `play` that takes the `self` parameter.

Reviewer 2: ```suggestion
You should define a method named `play` with a `self` parameter.
```
